src/components/Favorites/index.tsx

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import { FavoritesState } from "../../interfaces";
+import { RootState } from "../../redux/rootReducer";
+import "./Favorites.css";
+import FavoriteList from "./FavoriteList";
+
+const Favorites = () => {
+  const favoritesData: FavoritesState = useSelector(
+    (state: RootState) => state.favorites
```

Comment:
This function should be defined outside the component then simply:
`
const favoritesData = useSelector(getFavoritesData)
`

With this approach, you don't event need to define `FavoritesState` type as it will be inferred, additionally your selector function won't change it's reference.

Reply:
This is approach is similar to how you define actions for redux